src/main/java/com/pinterest/secor/main/ConsumerMain.java

```diff
@@ -54,6 +55,9 @@ public static void main(String[] args) {
         }
         try {
             SecorConfig config = SecorConfig.load();
+            if (!config.getUploadOnShutdown()) {
+                Runtime.getRuntime().addShutdownHook(new Thread(new StagingCleaner(config.getLocalPath())));
```

Comment:
Please use this path instead (Take a look at Upload.java):  config.getLocalPath() is a shared folder with other processes/threads running on that host.

                        String localPrefix = mConfig.getLocalPath() + '/' +
                            IdUtil.getLocalMessageDir();